File: samples/src/main/java/com/compdfkit/samples/SampleListAdapter.java
/**
 * Copyright © 2014-2025 PDF Technologies, Inc. All Rights Reserved.
 *
 * THIS SOURCE CODE AND ANY ACCOMPANYING DOCUMENTATION ARE PROTECTED BY INTERNATIONAL COPYRIGHT LAW
 * AND MAY NOT BE RESOLD OR REDISTRIBUTED. USAGE IS BOUND TO THE ComPDFKit LICENSE AGREEMENT.
 * UNAUTHORIZED REPRODUCTION OR DISTRIBUTION IS SUBJECT TO CIVIL AND CRIMINAL PENALTIES.
 * This notice may not be removed from this file.
 */

package com.compdfkit.samples;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatTextView;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class SampleListAdapter extends RecyclerView.Adapter<SampleListAdapter.SampleListViewHolder> {

    private List<PDFSamples> list = new ArrayList<>();

    private OnItemClickListener onItemClickListener;

    public SampleListAdapter(){
        list.addAll(SampleApplication.getInstance().samplesList);
    }

    @NonNull
    @Override
    public SampleListAdapter.SampleListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new SampleListViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_sample_list_item, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull SampleListAdapter.SampleListViewHolder holder, int position) {
        holder.tvSampleTitle.setText(list.get(holder.getAdapterPosition()).getTitle());
        holder.itemView.setOnClickListener(v -> {
            if (onItemClickListener != null) {
                onItemClickListener.onClick(holder.getAdapterPosition());
            }
        });
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
    }

    public static class SampleListViewHolder extends RecyclerView.ViewHolder{

        private AppCompatTextView tvSampleTitle;

        public SampleListViewHolder(@NonNull View itemView) {
            super(itemView);
            tvSampleTitle = itemView.findViewById(R.id.tv_sample_title);
        }
    }

    public interface OnItemClickListener{
        void onClick(int position);
    }
}

File: Samples_kotlin/src/main/java/com/compdfkit/samples/SampleListAdapter.kt
/**
 * Copyright © 2014-2025 PDF Technologies, Inc. All Rights Reserved.
 *
 * THIS SOURCE CODE AND ANY ACCOMPANYING DOCUMENTATION ARE PROTECTED BY INTERNATIONAL COPYRIGHT LAW
 * AND MAY NOT BE RESOLD OR REDISTRIBUTED. USAGE IS BOUND TO THE ComPDFKit LICENSE AGREEMENT.
 * UNAUTHORIZED REPRODUCTION OR DISTRIBUTION IS SUBJECT TO CIVIL AND CRIMINAL PENALTIES.
 * This notice may not be removed from this file.
 */
package com.compdfkit.samples

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.AppCompatTextView
import androidx.recyclerview.widget.RecyclerView

class SampleListAdapter : RecyclerView.Adapter<SampleListAdapter.SampleListViewHolder>() {

    private val list: MutableList<PDFSamples> = ArrayList()

    var onItemClickListener : ((position : Int) -> Unit)? = null

    init {
        list.addAll(SampleApplication.instance.samplesList)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SampleListViewHolder {
        return SampleListViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.layout_sample_list_item, parent, false))
    }

    override fun onBindViewHolder(holder: SampleListViewHolder, position: Int) {
        holder.tvSampleTitle.text = list[holder.adapterPosition].title
        holder.itemView.setOnClickListener {
            onItemClickListener?.invoke(holder.adapterPosition)
        }
    }

    override fun getItemCount(): Int {
        return list.size
    }

    class SampleListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvSampleTitle: AppCompatTextView

        init {
            tvSampleTitle = itemView.findViewById(R.id.tv_sample_title)
        }
    }

}